cli/commands/create/create.go

```diff
@@ -67,26 +71,57 @@ func execute(cli *cli.SensuCli) func(*cobra.Command, []string) error {
 	}
 }
 
+var jsonRe = regexp.MustCompile(`^(\s)*[\{\[]`)
+
+// parseResources is a rather heroic function that will parse any number of valid
+// JSON or YAML resources. Since it attempts to be intelligent, it likely
+// contains bugs.
+//
+// The general approach is:
+// 1. detect if the stream is JSON by sniffing the first non-whitespace byte.
+// 2. If the stream is JSON, goto 4.
+// 3. If the stream is YAML, split it on '---' to support multiple yaml documents.
+// 3. Convert the YAML to JSON document-by-document.
+// 4. Unmarshal the JSON one resource at a time.
 func parseResources(in io.Reader) ([]types.Resource, error) {
 	var resources []types.Resource
-	dec := json.NewDecoder(in)
-	dec.DisallowUnknownFields()
-	var err error
-	errCount := 0
-	for dec.More() {
-		var w types.Wrapper
-		if rerr := dec.Decode(&w); rerr != nil {
-			// Write out as many errors as possible before bailing,
-			// but cap it at 10.
-			err = errors.New("some resources couldn't be parsed")
-			if errCount > 10 {
-				err = errors.New("too many errors")
-				break
+	b, err := ioutil.ReadAll(in)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing resources: %s", err)
+	}
+	// Support concatenated yaml documents separated by '---'
+	array := bytes.Split(b, []byte("\n---\n"))
+	for _, b := range array {
+		var jsonBytes []byte
+		if jsonRe.Match(b) {
+			// We are dealing with JSON data
+			jsonBytes = b
+		} else {
+			// We are dealing with YAML data
```

Comment:
You split on `---` and _then_ you try to detect if each document is JSON or YAML.
This is not what the comment above `parseResources()` is describing.

I have no preference for either approach (split then detect JSON, or detect JSON then split), except for the fact that the current implementation makes it possible to have a mixed JSON/YAML input, which sounds... demonic.

Reply:
Haha. Well, the presence of `\n---\n` in a document indicates that it is not valid JSON. I've only done it that way for programmer convenience.

YAML is a superset of JSON -- so documents that contain a bunch of JSON blobs separated by --- are actually yaml. :)

It seems to me that if the split results in more than a single element, we should assume that the parser is dealing with yaml. I can add some logic around that.

Reply:
Thanks for the clarification!